Modify Rule S1226: Add exception for argument reassignments inside if statements (#1322)
SonarSource/rspec: rules/S1226/javascript/rule.adoc

}
----

== Exceptions

> There is a common pattern in Javascript to overwrite certain parameters depending on other parameters that are optional. For example, a callback is by convention always passed in the last position. If a parameter in a previous position was not passed, the callback will be passed in its position instead.

Reviewer: ```suggestion
There is a common pattern in JavaScript to overwrite certain parameters depending on other parameters that are optional. For example, a callback is by convention always passed in the last position. If a parameter in a previous position was not passed, the callback will be passed in its position instead.
```

}
----

== Exceptions

> There is a common pattern is Javascript to overwrite certain parameters depending on other optional arguments being passed. For example, a callback is by convention always passed in the last position, meaning that if a parameter is a previous position was not passed, the callback will be passed in its position instead.

Reviewer: ```suggestion
There is a common pattern in JavaScript to overwrite certain parameters depending on other parameters that are optional. For example, a callback is by convention always passed in the last position. If a parameter in a previous position was not passed, the callback will be passed in its position instead.
```

> Because of this pattern, parameter reassignments which are inside an `if` statement block will not raise an issue.

Reviewer: ```suggestion
Therefore, the rule ignores parameter reassignments that are inside an `if` statement block.
```

[source,javascript]
----

> // Use the initialState as a default value

Reviewer: I am not sure to understand how this comment is related to the function. Let's just drop it.
